Add support for specifying the RunAction `launchStyle`

baekteun opened the pull request:
Resolves https://github.com/tuist/tuist/issues/3511

### Short description 📝

Added a new property to RunAction called `launchStyle`

### How to test the changes locally 🧐

1. Generate a project with defined `launchStyle` for RunAction.
2. Go to `Edit scheme` -> `Run` and verify if the `Launch` On field is correctly set.

<img src="https://github.com/tuist/tuist/assets/74440939/ccb66266-1ca3-4cc1-a2a1-d451254f570e" width="500" />


### Contributor checklist ✅

- [x] The code has been linted using run `make lint-fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [x] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

danyf90 commented on Sources/ProjectDescription/RunAction.swift:
```diff
@@ -34,6 +34,9 @@ public struct RunAction: Equatable, Codable {
     public let expandVariableFromTarget: TargetReference?
 
+    /// The launch style of the action
+    public let launchStyle: LaunchStyle?
```
And set the default in the init
```suggestion
    public let launchStyle: LaunchStyle
```

danyf90 commented on Sources/TuistGraph/Models/RunAction.swift:
```diff
@@ -15,6 +15,7 @@ public struct RunAction: Equatable, Codable {
     public let options: RunActionOptions
     public let diagnosticsOptions: Set<SchemeDiagnosticsOption>
     public let expandVariableFromTarget: TargetReference?
+    public let launchStyle: LaunchStyle?
```
```suggestion
    public let launchStyle: LaunchStyle
```

danyf90 reviewed (commented):
Aren't we missing the mapping in `RunAction+ManifestMapper.swift`?

baekteun commented:
Thank you @danyf90 !

I pushed

- Changed the type of the launchStyle property of the RunAction structure from `LaunchStyle?` -> `LaunchStyle`.
- Changed default value of launchStyle from `nil` -> `.automatically` in functions that create RunAction or init

danyf90 commented:
@allcontributors add @baekteun for code

danyf90 commented on Sources/TuistGraph/Models/LaunchStyle.swift:
```diff
@@ -1,6 +1,16 @@
 public enum LaunchStyle: Codable {
     case automatically
     case waitForExecutableToBeLaunched
+
+    public static func from(launchStyle: ProjectDescription.LaunchStyle) -> TuistGraph.LaunchStyle {
+        switch launchStyle {
+        case .automatically:
+            return .automatically
+        case .waitForExecutableToBeLaunched:
+            return .waitForExecutableToBeLaunched
+        }
+    }
```
I would move this to `Sources/TuistLoader/Models+ManifestMappers/RunAction+ManifestMapper.swift` directly.
`TuistGraph` shouldn't depend on `ProjectDescription`

baekteun replied:
I tried placing the function that maps a ProjectDescription.LaunchStyle to TuistGraph.LaunchStyle in `TuistLoader/Models.ManifestMappers/LaunchStyle+ManifestMapper.swift`, is that okay?
